### ISSUE TITLE
min/max/mean/std return type Any instead of Tensor

### ISSUE THREAD

**Issue by adamjstewart** [pytorch/pytorch#116327]
### 🐛 Describe the bug

This bug started occurring with mypy 1.7+, so I'm not sure if this is a bug in PyTorch or a bug in mypy. To reproduce, create the following `test.py` file:
```python
import torch

x = torch.rand(2, 2)
reveal_type(x)
y = x.mean()
reveal_type(y)
```
Then run:
```console
> mypy --strict test.py
test.py:4: note: Revealed type is "torch._tensor.Tensor"
test.py:6: note: Revealed type is "Any"
```
The type of y should be Tensor as well. This is true in mypy 1.6 and older, and when not using strict mode.

### Versions

I don't use pip so the script doesn't work, but here are some relevant versions:

* Python 3.11.6
* PyTorch 2.1.2
* mypy 1.8.0

**Comment by Skylion007** [comment]
@adamjstewart Is this related to the new Type Inference algorithm in 1.7? https://github.com/python/mypy/blob/master/CHANGELOG.md#improved-type-inference can you try toggling it back to the old one and seeing if that is the cause.

**Comment by adamjstewart** [comment]
Nope, using `--old-type-inference` results in the same issue.

The other mypy issue I reported (#116328) is allegedly already fixed on main, so it's possible this one was too.

**Comment by Skylion007** [comment]
You can open a PR to update lintrunner's mypy version to confirm if this is the case.

**Comment by adamjstewart** [comment]
It looks like it's already running 1.7, so if there was an issue, we should know by now. I think we can probably close this. Thanks for your help!

**Comment by adamjstewart** [comment]
Can confirm that this issue was fixed in PyTorch 2.2!
